lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -642,6 +642,31 @@ sub _add_distri_and_version_to_summary {
     }
 }
 
+sub latest_jobs {
+    my ($self, $search_args, $until) = @_;
+    my @latest_jobs;
+
+    my $group_ids = $search_args->{groupids};
+    if (defined $group_ids && @$group_ids > 1) {
+        my @composed_search_arch = map {
+            my $local_search_args = {
+                group_id => $_,
+                build    => $self->schema->resultset('Jobs')->latest_build((groupid => $_))};
+            $local_search_args->{distri}  = $search_args->{distri}  if (defined($search_args->{distri}));
+            $local_search_args->{version} = $search_args->{version} if (defined($search_args->{version}));
```

Comment:
I'd save the search args within variables (e.g. `my ($distri, $version) = ($search_args->{distri}, $search_args->{version});`) instead of accessing these twice within each map iteration.

Reply:
I disagree. I don't see the need to copy hash values into variables. That's of course different if you save return values from functions which can be considered more costly

Reply:
I don't think Perl copies Strings, it'll just be a reference internally. This should be faster than the hash lookup. But I also like using variables from the point of DRY as it avoids repeating the key.

Reply:
I don't understand very well I should put that in variables. I need to build the hash to call the search method

Reply:
I mean the `$search_args` this function takes, **not** the hash this function creates (to call the search method).

Reply:
And btw, strings are COW in Perl, indeed: https://perldoc.perl.org/5.20.0/perldelta#Performance-Enhancements

But all of this won't really make much of a difference from the runtime performance so you can also keep your code if you like.